Makefile

```diff
@@ -20,7 +20,19 @@ OUTDIR ?= $(CURDIR)/out
 PKG=github.com/awslabs/soci-snapshotter
 VERSION=$(shell git describe --match 'v[0-9]*' --dirty='.m' --always --tags)
 REVISION=$(shell git rev-parse HEAD)$(shell if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi)
-GO_LD_FLAGS=-ldflags '-s -w -X $(PKG)/version.Version=$(VERSION) -X $(PKG)/version.Revision=$(REVISION) $(GO_EXTRA_LDFLAGS)'
+
+GO_BUILDTAGS ?=
+ifneq ($(STATIC),)
+	GO_BUILDTAGS += osusergo netgo static_build
+endif
+GO_TAGS=$(if $(GO_BUILDTAGS),-tags "$(strip $(GO_BUILDTAGS))",)
+
+GO_LD_FLAGS=-ldflags '-s -w -X $(PKG)/version.Version=$(VERSION) -X $(PKG)/version.Revision=$(REVISION) $(GO_EXTRA_LDFLAGS)
+ifneq ($(STATIC),)
```

Comment:
Can we merge some of these sections to be easier to read? 

Ideally we would only need one static check that sets several variables and then one assignment to `GO_LD_FLAGS`.

Reply:
Tags and linker flags are different things. So I think it's actually clearer to keep them separate as different variables.

Reply:
They don't need to be the same variable, I'd just like to see less context needed across lines. Something like:
```
BUILDTAGS ?=
OPTIONAL_LD_FLAGS=
 ifneq ($(STATIC),)
 	GO_BUILDTAGS += osusergo netgo static_build
        OPTIONAL_LD_FLAGS+=-extldflags "-static"
 endif
GO_TAGS=$(if $(GO_BUILDTAGS),-tags "$(strip $(GO_BUILDTAGS))",)
GO_LD_FLAGS=-ldflags '-s -w -X $(PKG)/version.Version=$(VERSION) -X $(PKG)/version.Revision=$(REVISION) $(OPTIONAL_LD_FLAGS) $(GO_EXTRA_LDFLAGS)'
```

Reply:
TBH I don't see how your approach is better/clearer than mine. You have to introduce another variable for the sake of combining the logic into a single if condition. which could lead to more readability issues once we have more flags to be customized in the future. Instead grouping logics per individual variable looks cleaner and more understandable, right?

Reply:
I guess I can buy keeping the if separate to group by tags and ldflags instead. The particular piece that I don't like about the current change is that Lines 30-34 are stateful. I think it's worth introducing a new variable there and interpolating with a single assignment to `GO_LD_FLAGS` rather than appending directly.

Reply:
It's a common pattern in Makefiles though. That's the reason of the `+=` operator I believe.

Reply:
We discussed a bit offline. I have this preference, but it's not a blocker to merging the change.